l3backend/l3backend-basics.dtx

```diff
@@ -90,6 +90,33 @@
   {l3backend-xdvipdfmx.def}{2020-06-29}{}
   {L3 backend support: xdvipdfmx}
 %</xdvipdfmx>
+%    \end{macrocode}
+%
+%   Check if the loaded kernel is at least enough to load this file.
+%   The kernel date has to be at least equal to \cs{ExplBackendFileDate}
+%   or later.  If \cs{__kernel_dependency_version_check:Nn} doesn't
+%   exist we're loading in an older kernel, so it's an error anyway.
+%   With time, this test should vanish and only the dependency check
+%   should remain.
+%    \begin{macrocode}
+\cs_if_exist:NTF \__kernel_dependency_version_check:nn
+  {
+    \__kernel_dependency_version_check:nn {2020-06-29}
```

Comment:
By what process/method is the date  in \__kernel_dependency_version_check:nn set?

Reply:
Manually.  As Joseph said in the [comment above](https://github.com/latex3/latex3/pull/787#commitcomment-41146960), the release cycle of `l3backend` is usually separate from `l3kernel`, so just setting the same date doesn't work, so the date has to be set manually to the version required.

Reply:
so I feared :-)  I wonder if that needs documentating (not sure where) and perhaps all moved to a separated filewhere the setting is all confined to one place.  I'm just wondering about how updates happen if they are done by somebody not doing them normally ... and how much fails because he/she doesn't know where to do adjustments.

Reply:
Perhaps a `l3version.dtx` somewhere that contains docstrip guards for all dependencies.  The problem is that these files are under different locations in the source (and distribution) trees, so I don't know how well (or bad) that would work out. Or maybe this can be updated semi-automatically with `l3build`, as the release date is. @josephwright probably knows the most details about this process... Would something like that be possible?

That said, the worst that would happen if some file had the wrong minimum date would be an error deeper down in the code as we already have, so back at square one in that case.